### ISSUE TITLE
Gdiffsplit cannot handle files named "config.py"

### ISSUE THREAD

**Issue by EgZvor** [tpope/vim-fugitive#1531]
This one's bizarre. `Gdiffsplit` and its variants hang when run on a file with a name `config.py`. Reproduced without any `vimrc` or plugins and regardless of directory structure.

```
$ git --version
git version 2.26.2
$ vim --version
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Apr 19 2020 22:46:53)
Included patches: 1-606
```

**Comment by EgZvor** [comment]
`Gedit` fails as well. And if I press `<cr>` during the hanging, it stops with

```
Error detected while processing function fugitive#BufReadCmd[89]..<SNR>15_ReplaceCmd:
line    6:
E325: ATTENTION
```

**Comment by EgZvor** [comment]
I traced this back (using `:h :breakadd` which I just found out about) to

```
function! s:ReplaceCmd(cmd) abort
  let temp = tempname()
  let [err, exec_error] = s:TempCmd(temp, a:cmd)
  if exec_error
    call s:throw((len(err) ? err : filereadable(temp) ? join(readfile(temp), ' ') : 'unknown error running ' . a:cmd))
  endif
  silent exe 'lockmarks keepalt 0read ++edit' s:fnameescape(temp)
  if &foldenable && foldlevel('$') > 0
    set nofoldenable
    silent keepjumps $delete _
    set foldenable
  else
    silent keepjumps $delete _
  endif
  call delete(temp)
  if s:cpath(fnamemodify(bufname('$'), ':p'), temp)
    silent! execute bufnr('$') . 'bwipeout'
  endif
endfunction
```
The hanging line was
```
silent exe 'lockmarks keepalt 0read ++edit' s:fnameescape(temp)
```
there was a swap file somewhere with a colliding name and so it asked what to do with it, waiting for input. I pressed `d` to delete, perhaps too hastily, and now the issue is gone.

**Comment by tpope** [comment]
That's weird, I wouldn't expect `:read` to create a swapfile, nor would I expect the swapfile for a temp file we just created to exist. What is `:set directory?`?

**Comment by EgZvor** [comment]
Right now it's `.,~/tmp,/var/tmp,/tmp`. I found some `.swp` files in `~/tmp` which may have been created from inside docker (maybe?), they have `600` permissions and my regular owner, but they're totally empty (0 bytes). For these files the behaviour is the same.

**Comment by tpope** [comment]
Next time it happens (or if you are able to reproduce it now), can you try adding `noswapfile` before `0read` and see if that fixes it?

**Comment by EgZvor** [comment]
Yep, that fixes it.

**Comment by tpope** [comment]
This was fixed when addressing #1598.
